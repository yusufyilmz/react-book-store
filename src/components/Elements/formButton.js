import React from 'react';
import {FormButtonWrapper} from './style';
import {Popup} from '../Popup';

export const FormButton = ({ text, showPopup, message }) => {
    return <FormButtonWrapper variant="primary" type="submit">
        {text}
        <Popup show={showPopup} message={message}/>
    </FormButtonWrapper>
}





