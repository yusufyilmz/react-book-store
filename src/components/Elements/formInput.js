import React from 'react';
import { Form } from 'react-bootstrap';

export const FormInput = ({title, name, id, onChange, type, placeholder }) => {
    return <Form.Group controlId={id}>
        <Form.Label>{title}</Form.Label>
        <Form.Control
            text={name}
            type={type}
            placeholder={placeholder}
            value={name}
            onChange={onChange}
            required />
    </Form.Group>
}




