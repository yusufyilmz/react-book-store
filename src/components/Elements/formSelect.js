

import React from 'react';
import { Form } from 'react-bootstrap';


const getOptions = (genres) => {

    return genres.map(genre => {
        return <option key={genre.id} value={genre.id}>{genre.name}</option>

    })
}


export const FormSelect = ({ title, id, value, onChange, options }) => {
    return <Form.Group controlId={id}>
        <Form.Label>{title}</Form.Label>
        <Form.Control placeholder="select" as="select" value={value} onChange={onChange}>
            {getOptions(options)}
        </Form.Control>
    </Form.Group>
}




