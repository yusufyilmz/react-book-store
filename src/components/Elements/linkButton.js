import React from 'react';
import { Row, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { ButtonWrapper } from './style';

export const LinkButton = ({ title, path, left, bottom, width }) => {
    return <Link to={path} >
        <ButtonWrapper
            left={left}
            bottom={bottom}
            width={width}
            variant="primary"
            type="button">
            {title}
        </ButtonWrapper>
    </Link>
}
