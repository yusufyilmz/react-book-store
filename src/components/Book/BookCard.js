import React from 'react';
import { BookHeaderWrapper, BookParagraphWrapper, InputGroupWrapper, CardWrapper } from './style';
import { Card, Col } from 'react-bootstrap';
import { LinkButton } from '../Elements/linkButton';


export const BookCard = ({ item }) => {
    return (
        <Col md={4}>
            <CardWrapper>
                <BookHeaderWrapper>{item.name}</BookHeaderWrapper>
                <Card.Body>
                    <BookParagraphWrapper>{item.genre.name}</BookParagraphWrapper>
                    <BookParagraphWrapper>{item.price} euro</BookParagraphWrapper>
                </Card.Body>
                <InputGroupWrapper>
                    <LinkButton
                        width="80"
                        path={`/books/edit/${item.id}`}
                        title="Edit" />
                    <LinkButton
                        width="80"
                        path={`/books/delete/${item.id}`}
                        title="Delete" />
                </InputGroupWrapper>
            </CardWrapper>
        </Col>

    );
}