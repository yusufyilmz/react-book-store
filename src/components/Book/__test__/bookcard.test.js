

import React from 'react';
import { BookCard } from '../BookCard';
import { shallow, mount } from 'enzyme'
import { LinkButton } from '../../Elements/linkButton';

describe('BOOKCARD COMPONENT', () => {

  
    var wrapper, props;

    beforeEach(() => {
        props = {
            item: {
                name: 'book 1',
                genre: {
                    name: 'comedy'
                },
                price: 1,
                id: 1
            }
        };
        wrapper = shallow(
            <BookCard {...props}/>
        );
    });


    test('renders element correctly', () => {
        expect(wrapper.find('CardWrapper').length).toBe(1)
    });
    
    test('renders text correctly', () => {
        expect(wrapper.find('BookParagraphWrapper').at(0).text()).toBe(props.item.genre.name)
    });
    
    test('renders props correctly', () => {
        expect(wrapper.find(LinkButton).at(0).prop('path')).toContain(props.item.id)
    });
  
    test('snapchat correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})
