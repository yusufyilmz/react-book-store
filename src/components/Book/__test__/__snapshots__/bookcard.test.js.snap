// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BOOKCARD COMPONENT snapchat correctly 1`] = `
<ForwardRef(Bootstrap(Col))
  md={4}
>
  <CardWrapper>
    <Styled(Component)>
      book 1
    </Styled(Component)>
    <ForwardRef(Bootstrap(CardBody))>
      <BookParagraphWrapper>
        comedy
      </BookParagraphWrapper>
      <BookParagraphWrapper>
        1
         euro
      </BookParagraphWrapper>
    </ForwardRef(Bootstrap(CardBody))>
    <Styled(Component)>
      <LinkButton
        path="/books/edit/1"
        title="Edit"
        width="80"
      />
      <LinkButton
        path="/books/delete/1"
        title="Delete"
        width="80"
      />
    </Styled(Component)>
  </CardWrapper>
</ForwardRef(Bootstrap(Col))>
`;
